api/services/orders/order_store.go

```diff
@@ -0,0 +1,40 @@
+package orders
+
+import (
+	"database/sql"
+	"log"
+
+	"github.com/sockify/sockify/types"
+)
+
+type SQLOrderStore struct {
+	db *sql.DB
+}
+
+// NewOrderStore initializes an order store that uses an SQL database
+func NewOrderStore(db *sql.DB) types.OrderStore {
+	return &SQLOrderStore{db: db}
+}
+
+// GetOrdersByStatus retrieves orders filtered by status from the database
+func (s *SQLOrderStore) GetOrdersByStatus(status string) ([]types.Order, error) {
+	var orders []types.Order
+
+	query := "SELECT * FROM orders WHERE status = $1"
+	rows, err := s.db.Query(query, status)
+	if err != nil {
+		log.Printf("Error fetching orders by status: %v", err)
+		return nil, err
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var order types.Order
+		if err := rows.Scan(&order.ID, &order.InvoiceNumber, &order.Address, &order.Contact, &order.CreatedAt, &order.Status); err != nil {
```

Comment:
The table doesn't actually have an `address` and `contact` column.

So, you have to create a `var address types.Address` and `var contact types.Contact` then scan the corresponding columns into the address and contact variables instead of `orders`.

Something like 

```go
var order types.Order
var contact types.Contact
var address types.Address

rows.Scan(
  &order.ID, &order.InvoiceNumber, &order.TotalPrice, &order.Status, 
  &contact.FirstName, &contact.Lastname..., 
  &address.Street, &address.AptUnit...
  &order.CreatedAt
);

order.Address = address
order.Contact = contact
```

These are all the columns in the order they come out the database 
<img width="675" alt="Screenshot 2024-10-03 at 8 25 09 AM" src="https://github.com/user-attachments/assets/21c21cc2-af05-4481-877e-e23b3621c5bb">